Crear modelo de Rent

paul-ot opened the pull request:
Trello Card: https://trello.com/c/dMtDZiZ8/20-crear-modelo-de-rent

- Creado el modelo Rent, con referencia a un User and Book.
- Puede crearse un rent y luego cada user acceder a sus propios rents.

paul-ot committed:
- Creado el modelo Rent, con referencia a un User and Book.
- Puede crearse un rent y luego cada user acceder a sus propios rents.

holywyvern commented on spec/factories/rents.rb:
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :rent do
+    user { nil }
```
Podes agregar constructores de otras factories, como user y book son requeridos, estaría bien agregar create(:user) y create(:book)

paul-ot replied:
Fixed.

holywyvern commented on spec/factories/rents.rb:
```diff
@@ -0,0 +1,8 @@
+  factory :rent do
+    user { nil }
+    book { nil }
+    start_date { "2019-02-06" }
```
Utiliza las fechas de faker: https://github.com/stympy/faker/blob/master/doc/v1.9.1/date.md

Podes incluso hacer `start_date` y luego hacer end date como `start_date + Faker::Number.between(1, 10).days` por ejemplo

paul-ot replied:
Fixed

holywyvern commented on spec/models/rent_spec.rb:
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Rent, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
No se si hay otra card de crear test de modelo Rent, pero, si no existe, deberías eliminar este archivo

paul-ot replied:
No la hay, borrado.

holywyvern commented on app/models/rent.rb:
```diff
@@ -0,0 +1,4 @@
+class Rent < ApplicationRecord
+  belongs_to :user
+  belongs_to :book
```
faltan las validaciones de las fechas

paul-ot replied:
Fixed, también añadí validaciones para que siempre sea necesario añadir `user_id` y `book_id`

holywyvern commented on app/models/user.rb:
```diff
@@ -9,4 +9,6 @@ class User < ApplicationRecord
   validates :first_name, presence: true
   validates :last_name, presence: true
+
+  has_many :rents
```
Esto tambien debería estar en Book, como `has_many :rents`

paul-ot replied:
Fixed.

paul-ot committed:
- Added relationship between Book and Rent models.
- Removed spec/models/rent_spec.rb, since we're not going to use it.
- Added field validations in Rent model.
- Better fields values in the rent factory.

paul-ot commented:
Listos los cambios pedidos, podrían revisar nuevamente?
@holywyvern 
Gracias!
